CMakeLists.txt

```diff
@@ -1,68 +1,35 @@
 cmake_minimum_required(VERSION 3.2)
 
 # TODO: SDK Versioning.
-project(ggpo VERSION 1.0.0)
+project(GGPO VERSION 1.0.0)
 
-# What do we want to build?
-option(BUILD_GGPO "Enable the build of the GGPO SDK" ON)
-option(BUILD_VECTORWAR "Enable the build of the Vector War example app" ON)
-option(BUILD_SHARED_LIBS "Enable the build of shared libraries (.dll/.so) instead of static ones (.lib/.a)" ON)
+# Remove RelWithDebInfo and MinSizeRelease build types to match the original Visual Studio Project. 
+set(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "" FORCE)
 
-set(GGPO_SDK
-	src/include/ggponet.h
-	src/lib/ggpo/main.cpp
-	src/lib/ggpo/bitvector.h
-	src/lib/ggpo/bitvector.cpp
-	src/lib/ggpo/game_input.h
-	src/lib/ggpo/game_input.cpp
-	src/lib/ggpo/input_queue.h
-	src/lib/ggpo/input_queue.cpp
-	src/lib/ggpo/log.h
-	src/lib/ggpo/log.cpp
-	src/lib/ggpo/poll.h
-	src/lib/ggpo/poll.cpp
-	src/lib/ggpo/ring_buffer.h
-	src/lib/ggpo/static_buffer.h
-	src/lib/ggpo/sync.h
-	src/lib/ggpo/sync.cpp
-	src/lib/ggpo/timesync.h
-	src/lib/ggpo/timesync.cpp
-	src/lib/ggpo/types.h
-	src/lib/ggpo/zconf.h
-	src/lib/ggpo/zlib.h
-	src/lib/ggpo/backends/backend.h
-	src/lib/ggpo/backends/p2p.h
-	src/lib/ggpo/backends/p2p.cpp
-	src/lib/ggpo/backends/spectator.h
-	src/lib/ggpo/backends/spectator.cpp
-	src/lib/ggpo/backends/synctest.h
-	src/lib/ggpo/backends/synctest.cpp
-	src/lib/ggpo/network/udp.h
-	src/lib/ggpo/network/udp.cpp
-	src/lib/ggpo/network/udp_msg.h
-	src/lib/ggpo/network/udp_proto.h
-	src/lib/ggpo/network/udp_proto.cpp
-)
+include(src/cmake/properties.cmake)
+include(src/cmake/helper_methods.cmake)
 
-if(BUILD_GGPO)
-	add_library(ggpo ${GGPO_SDK})
-	
-	target_include_directories(ggpo PUBLIC src/include)
-	target_include_directories(ggpo PUBLIC src/lib/ggpo)
-	
-	if(WIN32 AND BUILD_SHARED_LIBS)
-		# Link to Multimedia API and Winsocks during a static build.
-		target_link_libraries(ggpo LINK_PUBLIC winmm.lib ws2_32.lib)
+if(WIN32)
+	## Prevent the CMake trying to install GGPO in Program Files on Windows
+	if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+		set (CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/installed" CACHE PATH "Default install path" FORCE)
 	endif()
-	
-	set_target_properties(ggpo PROPERTIES VERSION ${PROJECT_VERSION})
 endif()
 
-if(BUILD_VECTORWAR)
-	# Vector War is windows only.
+# What do we want to build?
+option(GGPO_BUILD_SDK "Enable the build of the GGPO SDK" ON)
+option(GGPO_BUILD_VECTORWAR "Enable the build of the Vector War example app" ON)
+option(GGPO_BUILD_SHARED_LIBS "Enable the build of shared libraries (.dll/.so) instead of static ones (.lib/.a)" ON)
```

Comment:
Is there a reason why we need to use a re-definition of BUILD_SHARED_LIBS? It means we have to explicitly state the library type when calling add_library().

Reply:
Yeah, I didn't know that ```BUILD_SHARED_LIBS``` was a built-it option and I wanted to be more explicit (and it was late when I was finishing this). Will change that.

Reply:
Done. Now ```BUILD_SHARED_LIBS``` just inherits the ```GGPO_BUILD_SHARED_LIBS``` value. I've also reverted the changes to the add_library and ```BUILD_SHARED_LIBS``` stuff to what was before.

Reply:
Why not just remove ```GGPO_BUILD_SHARED_LIBS``` altogether? Seeing as ```BUILD_SHARED_LIBS``` is builtin, having the two options linked together could be confusing, especially if one is set explicitly on the cli making them out of sync.

Reply:
Wanted to make the naming uniform with the other options, but I can understand the concern. Change reverted.